README
This is the Git repository for the TYPO3 extension EXT:beautyofcode

TYPO3 TER: http://typo3.org/extensions/repository/view/beautyofcode
Bugtracker: http://forge.typo3.org/projects/extension-beautyofcode

ChangeLog
Since version 3.2.0 see GitHub: https://github.com/fnagel/beautyofcode/commits/master

2016-12-11 Felix Nagel <info@felixnagel.com>

	3.1.0 released (no changes required)
	* TYPO3 8.4 compatibility
	* Bugfixes for t3editor EM configuration
	* Bugfix for new content element wizard
	* Code is now PSR-2 standard compliant

2016-01-25 Felix Nagel <info@felixnagel.com> & Thomas Juhnke <typo3@van-tomas.de>

	3.0.0 released (Migration needed see documentation)
	* TYPO3 7.6 compatibility, removed 6.2 support
	* Changed TypoScript processing
	* Added and fixed some languages for PrismJs
	* Update PrismJs vendor
	* Add composer and bower file
	* Now using default tt_content bodytext field for code

2015-08-10 Felix Nagel <info@felixnagel.com> & Thomas Juhnke <typo3@van-tomas.de>

	2.1.0 released (no changes required)
	* TYPO3 7.x compatibility

2014-08-09 Felix Nagel <info@felixnagel.com> & Thomas Juhnke <typo3@van-tomas.de>

	2.0.1 released (no changes required)
	* Improved highlighter configuration handling
	* Possible to switch highlighter engine without updating flexform
	* Remove Prism language add-ons from flexform select
	* Improved description for TS constants
	* Use addJsFooterFile instead of addJsFooterLibrary as default to improve FE performance
	* Improve readme file
	* Improve documentation (clean up, new screenshots)

2014-07-24 Felix Nagel <info@felixnagel.com> & Thomas Juhnke <typo3@van-tomas.de>

	2.0.0 released (Migration needed see documentation)
	* Rewritten for Extbase/Fluid usage
	* Refactoring of Backend Hooks (Flexform, T3editor Wizard)
	* Introduction of PageRenderer ViewHelpers for asset injection
	* Introduction of (basic) Unit tests
	* Integration of Prism (see http://prismjs.com/)
	* Removal of SyntaxHighlighter v2 + beautyofcode (jQuery) abstraction layer
	* Introducing ReST documentation

	Note: This release is introducing breaking changes. Please make sure to
	make a backup of your current site content and use the update script to update
	your boc content elements.

2013-09-05 Felix Nagel  <info@felixnagel.com>

	1.0.3 released (no changes required)
	* fixed: label subpart rendering, see #51541, thx to Tommy Juhnke
	* improved: general coding style and CGL

2013-03-28 Felix Nagel  <info@felixnagel.com>

	1.0.2 released (no changes required)
	* fixed: issue with adding jQuery file to the document, see #46755, thx to Jochen Fröhlich

2013-03-11 Felix Nagel  <info@felixnagel.com>

	1.0.1 released (no changes required)
	* fixed: requirements as needed for planned TYPO3 TER cleanup

2013-01-26 Felix Nagel  <info@felixnagel.com>

	1.0.0 released (you need to clear all caches)
	* added: add possibility to change the on ready callback signature of jQuery, thx to Thomas Juhnke, see http://forge.typo3.org/issues/44762

2012-11-26 Felix Nagel  <info@felixnagel.com>

	* improved: t3editor integration in BE, now code language aware (at least for PHP, CSS, JS, TS, XML)
	* removed: old t3editor integration files (EXT:typoscript_code)
	* improved: BE layout title
	* fixed: CGL
	* added: inline TODO comments
	* removed: outdated documentation pdf
	* fixed: copyright year

2012-09-06 Felix Nagel  <info@felixnagel.com>

	0.8.4 released (no changes required)
	* changed: tested with TYPO3 6.0.0 beta1
	* removed: deprecated functions
	* changed: extension state set to stable

2012-08-25 Felix Nagel  <info@felixnagel.com>
	* changed: files and inline code is now added using pageRenderer functions
	* changed: TYPO3 4.3 is now minimal requirement

2011-05-30 	Felix Nagel  <info@felixnagel.com>
	0.8.3 released (no changes required)
	* fixed: re-enabled t3editor in BE for TYPO 4.5.x, thx to EXT:typoscript_code

2011-05-26 	Felix Nagel  <info@felixnagel.com>
	0.8.2 released (no changes required)
	* added: API documentation (EXT:boc_apiexample)
	* fixed: small issue when using API

2011-03-13 	Felix Nagel  <info@felixnagel.com>
	0.8.1 released (no changes required)
	* fixed: disabled t3editor in BE for TYPO 4.5

2011-01-14 	Felix Nagel  <info@felixnagel.com>
	0.8 released (no changes required)
	* added: API
	* improved: BE preview

2011-01-07 	Felix Nagel  <info@felixnagel.com>
	* fixed: t3editor follow-up
	* fixed: missing parameter class instance (errors in sys log)
	* removed: REAMDE.txt
	* updated: documentation

2010-12-26 	Felix Nagel  <info@felixnagel.com>
	* added: t3editor syntax highlighting in BE
	* fixed: small locallang issue (German)
	* updated: documentation

2010-11-25  Felix Nagel  <info@felixnagel.com>
	0.7.1 released (no changes required)
	* added: Typoscript brush for v3
	* fixed: jQueryNoConflict has no effect, thx to Jürgen Furrer
	* updated: documentation

2010-11-06  Felix Nagel  <info@felixnagel.com>
	0.7.0 released (big one, changes required)
	* added: non-jquery support
	* removed: extension manager settings
	* added: options fpr standalone version
	* updated: SyntaxHighlighter to v3
	* updated: documentation
	* added: default CSS code

2010-11-05  Felix Nagel  <info@felixnagel.com>
	0.6.3 released (no changes required)
	* fixed: changlog and upload comment

2010-11-05  Felix Nagel  <info@felixnagel.com>
	0.6.2 released (no changes required)
	* fixed: error in constant editor (tenmplate file path)
	* fixed: TS config defaults does not work

2010-10-05  Felix Nagel  <info@felixnagel.com>
	* fixed: defaults.tab-size constant is not used

2010-10-05  Felix Nagel  <info@felixnagel.com>
	0.6.1 released (no changes required except you're using XCLASS)
	* added: hooks to add custom JS and template marker
	* changed: updated screenshots in doc
	* added: Doc FAQ: error solution
	* added: Doc Tutorial: how to add a template marker
	* fixed: missing config templateFile var in doc

2010-09-03  Felix Nagel  <info@felixnagel.com>
	0.6.0 released (small changes required)
	* added: it's now possible to use HTML templates
	* changed: label.show renamed to showLabel
	* changed: TYPO3 API instead of homemade functions
	* fixed: errors in changelog

ext_emconf.php
<?php

/***************************************************************
 * Extension Manager/Repository config file for ext "beautyofcode".
 *
 * Auto generated 05-09-2013 20:25
 *
 * Manual updates:
 * Only the data in the array - everything else is removed by next
 * writing. "version" and "dependencies" must not be touched!
 ***************************************************************/

$EM_CONF[$_EXTKEY] = array(
    'title' => 'beautyOfCode Syntax Highlighter',
    'description' => 'This plugin provides Java-Script based, state-of-the-art, feature rich syntax highlighting by using SyntaxHighlighter or Prism. t3editor enabled.',
    'category' => 'plugin',
    'shy' => 0,
    'version' => '3.3.1-mf',
    'priority' => '',
    'loadOrder' => '',
    'module' => '',
    'state' => 'stable',
    'uploadfolder' => 0,
    'createDirs' => '',
    'modify_tables' => '',
    'clearcacheonload' => 0,
    'lockType' => '',
    'author' => 'Felix Nagel',
    'author_email' => 'info@felixnagel.com',
    'constraints' => array(
        'depends' => array(
            'php' => '5.4.0-7.1.99',
            'typo3' => '7.3.1-8.7.99',
        ),
        'conflicts' => array(
        ),
        'suggests' => array(
            't3editor' => '',
        ),
    ),
    'autoload' => array(
        'psr-4' => array(
            'TYPO3\\Beautyofcode\\' => 'Classes',
        ),
    ),
);
